compiler/src/dashboard_compiler/cli.py

```
@@ -1100,9 +1100,16 @@ def disassemble(input_file: Path | None, output: Path) -> None:
         if input_file is None:
+            import io
             import sys
 
-            content = sys.stdin.read()
+            # Use TextIOWrapper to ensure UTF-8 encoding when reading from stdin
+            # This avoids issues on Windows where the default encoding might not be UTF-8
+            if hasattr(sys.stdin, 'buffer'):
+                content = io.TextIOWrapper(sys.stdin.buffer, encoding='utf-8').read()
+            else:
+                # Fallback for environments where stdin.buffer is not available (e.g. some tests)
+                content = sys.stdin.read()
```

Ruff (0.14.10)

Use ternary operator `content = io.TextIOWrapper(sys.stdin.buffer, encoding='utf-8').read() if hasattr(sys.stdin, 'buffer') else sys.stdin.read()` instead of `if`-`else`-block

(SIM108)